File: Leetcode/Leetcode DSA course/Arrays and Strings/Prefix Sum/3-2090 K Radius Subarray Averages/main.ts
function getAverages(nums: number[], k: number): number[] {
  if (k == 0) return nums;
  const n = nums.length;
  const windowSize = 2 * k + 1;
  const averages: number[] = new Array(n).fill(-1);

  if (windowSize > n) return averages;

  let windowSum = 0;
  for (let i = 0; i < windowSize; i++) windowSum += nums[i];

  averages[k] = Math.floor(windowSum / windowSize);

  for (let i = windowSize; i < n; i++) {
    windowSum += nums[i] - nums[i - windowSize];
    averages[i - k] = Math.floor(windowSum / windowSize);
  }

  return averages;
}

File: Leetcode/Leetcode DSA course/Arrays and Strings/Prefix Sum/3-2090 K Radius Subarray Averages/main.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        if (k == 0)
            return nums;
        int n = nums.size();
        int windowSize = 2 * k + 1;
        vector<int> avgs(nums.size(), -1);
        if (windowSize > n)
            return avgs;

        long long windowSum = 0;
        for (int i = 0;i < windowSize;i++)
            windowSum += nums[i];
        avgs[k] = windowSum / windowSize;

        for (int i = windowSize;i < n;i++)
        {
            windowSum += nums[i] - nums[i - windowSize];
            avgs[i - k] = windowSum / windowSize;
        }

        return avgs;
    }
};